<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:macro name="external_rrbot_force_torque_sensor" params="name prefix use_mock_hardware:=^|true mock_sensor_commands:=^|false">

    <ros2_control name="${name}" type="sensor">
      <hardware>
        <xacro:if value="${use_mock_hardware}">
          <plugin>mock_components/GenericSystem</plugin>
          <param name="mock_sensor_commands">${mock_sensor_commands}</param>
          <param name="position_state_following_offset">0.0</param>
        </xacro:if>
        <xacro:unless value="${use_mock_hardware}">
          <plugin>ros2_control_demo_example_5/ExternalRRBotForceTorqueSensorHardware</plugin>
          <param name="example_param_hw_start_duration_sec">0.0</param>
          <param name="example_param_hw_stop_duration_sec">3.0</param>
          <param name="example_param_max_sensor_change">5.0</param>
        </xacro:unless>
      </hardware>

      <sensor name="${prefix}tcp_fts_sensor">
        <state_interface name="force.x"/>
        <state_interface name="force.y"/>
        <state_interface name="force.z"/>
        <state_interface name="torque.x"/>
        <state_interface name="torque.y"/>
        <state_interface name="torque.z"/>
        <param name="frame_id">tool_link</param>
        <param name="fx_range">100</param>
        <param name="fy_range">100</param>
        <param name="fz_range">100</param>
        <param name="tx_range">15</param>
        <param name="ty_range">15</param>
        <param name="tz_range">15</param>
      </sensor>
    </ros2_control>
  </xacro:macro>

</robot>
